internal/services/managedapplications/managed_application_resource.go

```diff
@@ -354,37 +366,64 @@ func flattenManagedApplicationParametersOrOutputs(input *interface{}) (map[strin
 		for k, val := range attrs.(map[string]interface{}) {
 			mapVal, ok := val.(map[string]interface{})
 			if !ok {
-				return nil, fmt.Errorf("unexpected managed application parameter or output type: %+v", mapVal)
+				return nil, fmt.Errorf("unexpected managed application parameter type: %+v", mapVal)
 			}
 			if mapVal != nil {
 				v, ok := mapVal["value"]
 				if !ok {
-					return nil, fmt.Errorf("missing key 'value' in parameters or output map %+v", mapVal)
-				}
-				switch t := v.(type) {
-				case float64:
-					results[k] = v.(float64)
-				case string:
-					results[k] = v.(string)
-				case map[string]interface{}:
-					// Azure NVA managed applications read call returns empty map[string]interface{} parameter 'tags'
-					// Do not return an error if the parameter is unsupported type, but is empty
-					if len(v.(map[string]interface{})) == 0 {
-						log.Printf("parameter '%s' is unexpected type %T, but we're ignoring it because of the empty value", k, t)
-					} else {
-						return nil, fmt.Errorf("unexpected parameter type %T", t)
+					// Secure values are not returned, thus settings it with local value
+					v = ""
+					if oldValueStruct, oldValueStructOK := localParameters[k]; oldValueStructOK {
+						if oldValue, oldValueOK := oldValueStruct.(map[string]interface{})["value"]; oldValueOK {
```

Comment:
I think we'll need some type-checking here because `oldValueStruct` is an `interface{}` and no guarantee that it will be a map? If it isn't, this will panic

Reply:
`oldValueStruct` comes from expanded blocks from terraform sdk, not from api side so we shall be able to asure it is a map.

Reply:
yes make sense, have added the type check